azurerm/internal/services/containers/container_group_resource.go

```diff
@@ -942,12 +970,17 @@ func expandContainerVolumes(input interface{}) (*[]containerinstance.VolumeMount
 
 		cv := containerinstance.Volume{
 			Name: utils.String(name),
-			AzureFile: &containerinstance.AzureFileVolume{
+		}
+
+		if gitRepoVolume := expandGitRepoVolume(volumeConfig["git_repo"].([]interface{})); gitRepoVolume != nil {
```

Comment:
We'll need to do some logic to make sure that either `git_repo` is specified or the share name/storage account information is specified. We'll also want to check that both aren't specified. The last check will be that all the share name and storage account information is specified if `git_repo` isn't specified. If one of those checks fail, we'll need to raise that error for the user.